adafruit_si7021.py

```diff
@@ -19,11 +19,140 @@
+        for i in range(8):
+            if crc & 0x80:
+                crc <<= 1
+                crc ^= 0x131
+            else:
+                crc <<= 1
+    return crc
+
+
+class SI7021:
+    """
+    A driver for the SI7021 temperature and humidity sensor.
+    """
+
+    def __init__(self, i2c, address=0x40):
+        self.i2c_device = I2CDevice(i2c, address)
+        self.init()
+        self._measurement = 0
+
+    def init(self):
+        self.reset()
+        # Make sure the USER1 settings are correct.
+        while True:
+            # While restarting, the sensor doesn't respond to reads or writes.
+            try:
+                data = bytearray([_READ_USER1])
+                with self.i2c_device as i2c:
+                    i2c.writeto(data, stop=False)
+                    i2c.readfrom_into(data)
+                value = data[0]
+            except OSError as e:
+                if e.args[0] not in ('I2C bus error', 19): # errno 19 ENODEV
+                    raise
+            else:
+                break
+        if value != _USER1_VAL:
+            raise RuntimeError("bad USER1 register (%x!=%x)" % (
+                value, _USER1_VAL))
+
+    def _command(self, command):
+        with self.i2c_device as i2c:
+            i2c.writeto(ustruct.pack('B', command))
+
+    def _data(self):
+        data = bytearray(3)
+        data[0] = 0xff
+        while True:
+            # While busy, the sensor doesn't respond to reads.
+            try:
+                with self.i2c_device as i2c:
+                    i2c.readfrom_into(data)
+            except OSError as e:
+                if e.args[0] not in ('I2C bus error', 19): # errno 19 ENODEV
+                    raise
+            else:
+                if data[0] != 0xff: # Check if read succeeded.
+                    break
+        value, checksum = ustruct.unpack('>HB', data)
+        if checksum != _crc(data[:2]):
+            raise ValueError("CRC mismatch")
+        return value
+
+    def reset(self):
+        self._command(_RESET)
+
+    @property
+    def relative_humidity(self):
+        """
+        The measured relative humidity in percents.
+
+        If no measurement was started, this will start it automatically.
+        In either case, it will wait for the measurement's completion and
+        return the result.
+        """
+        self.start_measurement(HUMIDITY)
+        value = self._data()
+        self._measurement = 0
+        return value * 125 / 65536 - 6
+
+    @property
+    def temperature(self):
+        """
+        The measured temperature in degrees Celcius.
+
+        If no measurement was started, this will start it automatically.
+        In either case, it will wait for the measurement's completion and
+        return the result.
+        """
+        self.start_measurement(TEMPERATURE)
+        value = self._data()
+        self._measurement = 0
+        return value * 175.72 / 65536 - 46.85
+
+    def start_measurement(self, what):
```

Comment:
According to the datasheet, the conversion time can take up to 12ms. That is already noticeable by a naked eye when playing an animation, or by a naked ear when playing a melody.

I think it's useful to leave this function (obviously, otherwise I wouldn't leave it), even if most users will not use it. Unless you have a better idea about how to handle this? Perhaps the async module?